src/rez/package_test.py

```diff
@@ -718,3 +841,163 @@ def print_summary(self):
 
         strs = columnise(rows)
         print('\n'.join(strs))
+
+
+def _is_conflict(*requires):
+    """Check if any of the requirements have a conflict.
+
+    Example:
+        >>> _is_conflict([PackageRequest("foo-2")], [PackageRequest("!foo-2")])
+        # True
+        >>> _is_conflict([PackageRequest("foo-2")], [PackageRequest("foo-1+")])
+        # False
+
+    Args:
+        requires (tuple[iter[PackageRequest]]):
+            A series of Rez packages requests to check for conflicts.
+
+    Returns:
+        bool: If conflict, return True, otherwise, return False.
+
+    """
+    requirement = RequirementList(itertools.chain(*requires))
+
+    return requirement.conflict
+
+
+def _choose_test_object(package, test_requires, extra_packages=frozenset()):
+    """Find the most appropriate Rez object for ``rez-test --interactive``.
+
+    If ``package`` has variants, find a valid variant, given ``test_requires``
+    and ``extra_packages``.
+
+    Args:
+        package (Package):
+            The installed Rez package to query variant / test data from.
+        test_requires (set[PackageRequest], optional)
```

Comment:
It's a positional parameter, so it's required, not optional. Also you don't need to document if a field is optional or not.

Reply:
Ah I missed that one after the refactor, fixed. I'll do another pass to make sure I got them all. It's left over from when I previously had --interactive resolve multiple tests.

Though it's not required, stating optional is supported by sphinx.ext.napoleon and Sphinx [parses the type](https://crimsonscholar.github.io/rez_sphinx/latest/api/rez_sphinx.core.configuration.html#rez_sphinx.core.configuration.ConfPy.get_attributes) correctly.

Probably moot since one day we'll probably end up using type hints for this stuff anyway.